api/handlers/jobs.ts

```diff
@@ -0,0 +1,259 @@
+}
+
+// These options should only be defined if the build summary is being called after
+// a Gatsby Cloud job
+interface BuildSummaryOptions {
+  mongoClient?: MongoClient;
+  previewUrl?: string;
+}
+
+export const extractUrlFromMessage = (fullDocument): string[] => {
+  const { logs } = fullDocument;
+  const urls = logs?.length > 0 ? logs.flatMap((log) => log.match(/\bhttps?:\/\/\S+/gi) || []) : [];
+  return urls.map((url) => url.replace(/([^:]\/)\/+/g, '$1'));
+};
+
+function prepGithubComment(fullDocument: Job | EnhancedJob, jobUrl: string, isUpdate = false): string {
+  if (isUpdate) {
+    return `\n* job log: [${fullDocument.payload.newHead}](${jobUrl})`;
+  }
+  const urls = extractUrlFromMessage(fullDocument);
+  let stagingUrl = '';
+  if (urls.length > 0) {
+    stagingUrl = urls[urls.length - 1];
+  }
+  return `✨ Staging URL: [${stagingUrl}](${stagingUrl})\n\n#### 🪵 Logs\n\n* job log: [${fullDocument.payload.newHead}](${jobUrl})`;
+}
+
+async function prepSummaryMessage(
+  env: string,
+  fullDocument: Job,
+  repoName: string,
+  jobUrl: string,
+  jobId: string,
+  failed = false,
+  previewUrl?: string
+): Promise<string> {
+  const urls = extractUrlFromMessage(fullDocument);
+  let mms_urls: Array<string | null> = [null, null];
+  // mms-docs needs special handling as it builds two sites (cloudmanager & ops manager)
+  // so we need to extract both URLs
+  if (repoName === 'mms-docs') {
+    if (urls.length >= 2) {
+      mms_urls = urls.slice(-2);
+    }
+  }
+
+  let url = '';
+  if (previewUrl) {
+    url = previewUrl;
+  } else if (urls.length > 0) {
+    url = urls[urls.length - 1];
+  }
+
+  let msg = '';
+  if (failed) {
+    msg = `Your Job <${jobUrl}${jobId}|Failed>! Please check the build log for any errors.\n- Repo: *${repoName}*\n- Branch: *${fullDocument.payload.branchName}*\n- urlSlug: *${fullDocument.payload.urlSlug}*\n- Env: *${env}*\n Check logs for more errors!!\nSorry  :disappointed:! `;
+  } else {
+    if (repoName == 'mms-docs') {
+      msg = `Your Job <${jobUrl}${jobId}|Completed>! \n- Repo: *${repoName}*\n- Branch: *${fullDocument.payload.branchName}*\n- Commit: *${fullDocument.payload.newHead}*\n- urlSlug: *${fullDocument.payload.urlSlug}*\n- Env: *${env}*\n*Urls*\n   *CM*: <${mms_urls[0]}|Cloud Manager> \n   *OPM*: <${mms_urls[1]}|OPS Manager>\n- InvalidationStatus: <${fullDocument.invalidationStatusURL}|Status> \nEnjoy  :smile:!`;
+    } else {
+      msg = `Your Job <${jobUrl}${jobId}|Completed>! \n- Repo: *${repoName}*\n- Branch: *${fullDocument.payload.branchName}*\n- Commit: *${fullDocument.payload.newHead}*\n- urlSlug: *${fullDocument.payload.urlSlug}*\n- Env: *${env}*\n- Url: <${url}|${repoName}>\n- InvalidationStatus: <${fullDocument.invalidationStatusURL}|Status> \nEnjoy  :smile:!`;
+    }
+  }
+  // Remove instances of two or more periods
+  return msg.replace(/\.{2,}/g, '');
+}
+
+export async function notifyBuildSummary(jobId: string, options: BuildSummaryOptions = {}): Promise<any> {
```

Comment:
I believe this is a good thing! I am assuming that we will want this behavior for the enhanced autobuilder if we currently do this in `prd`

Reply:
Oh, duh, for some reason I was thinking the GitHub commenting logic would be called twice. Forgot that it would be isolated to a single call based on the webhook currently used by the author. Resolving this thread 😄